File: 2021/day-1/part-1/main.py
#!/usr/bin/env python3

import sys

def read_input(filename):
    input = []
    with open(filename) as fd:
        lines = fd.readlines()
        for line in lines:
            input.append(int(line.strip()))
    return input

def main():
    input = read_input("input.txt")
    result = 0
    for i in range(0, len(input)-1):
        if input[i] < input[i+1]:
            result += 1
    print(result)

if __name__ == "__main__":
    sys.exit(main())
            

File: 2021/day-1/part-1/main.jl
function main()
    local result = 0
    local content = readlines("input.txt")
    local lines = parse.(Int, content)
    for i in 2:length(lines)
        if lines[i] > lines[i-1]
            result += 1
        end
    end
    println(result)
end

main()